var/spack/repos/builtin/packages/blaspp/package.py

```diff
@@ -4,90 +4,49 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+import os
 
 
 class Blaspp(CMakePackage, CudaPackage):
     """C++ API for the Basic Linear Algebra Subroutines. Developed by the
        Innovative Computing Laboratory at the University of Tennessee,
        Knoxville."""
 
-    homepage = "https://bitbucket.org/icl/blaspp"
-    git      = "https://bitbucket.org/icl/blaspp"
+    homepage = 'https://bitbucket.org/icl/blaspp'
+    git = homepage
+    url = 'https://bitbucket.org/icl/blaspp/downloads/blaspp-2020.09.00.tar.gz'
     maintainers = ['teonnik', 'Sely85', 'G-Ragghianti', 'mgates3']
 
-    version('develop', commit='6293d96')
+    version('master', branch='master')
+    version('2020.10.00', sha256='ce148cfe397428d507c72d7d9eba5e9d3f55ad4cd842e6e873c670183dcb7795')
+    version('2020.09.00', sha256='ee5d29171bbed515734007dd121ce2e733e2f83920c4d5ede046e657f4a513ef')
 
-    variant('gfort',
-            default=False,
-            description=('Use GNU Fortran interface. '
-                         'Default is Intel interface. (MKL)'))
-    variant('ilp64',
-            default=False,
-            description=('Use 64bit integer interface. '
-                         'Default is 32bit. (MKL & ESSL)'))
-    variant('openmp',
-            default=False,
-            description=('Use OpenMP threaded backend. '
-                         'Default is sequential. (MKL & ESSL)'))
+    variant('openmp', default=True, description='Use OpenMP internally.')
+    variant('cuda',   default=True, description='Build with CUDA')
+    variant('shared', default=True, description='Build shared libraries')
 
+    depends_on('cmake@3.15.0:', type='build')
     depends_on('blas')
 
-    # 1) The CMake options exposed by `blaspp` allow for a value called `auto`.
-    #    The value is not needed here as the choice of dependency in the spec
-    #    determines the appropriate flags.
-    #
-    # 2) BLASFinder.cmake handles most options. For `auto`, it searches all
-    #    blas libraries listed in `def_lib_list`.
-    #
-    # 3) ?? Custom blas library can be supplied via `BLAS_LIBRARIES`.
-    #
+    # This will attempt to use a supported version of OpenBLAS
+    depends_on('openblas@:0.3.5', when='^openblas')
```

Comment:
@mgates @G-Ragghianti do you remember for which combinations/platforms you saw issues with BlasPP and newer OpenBLAS releases?

In all my (trivial) environment cases, the Spack concretizer fails to resolve the upper bound specifified in this package and my users need to manually add `^openblas@0.3.5` to concretize - but when I just comment out the conflict I see no issues (on Ubuntu).

Reply:
Also, OpenBLAS has compile issues on Ubuntu 20.04 ...

Reply:
Just commenting out the upper bound on OpenBLAS 0.3.5 and conflict works for me, though... Should we remove the constrain?

Reply:
I think the concretization issue comes from depending on the virtual `blas` package but contraining on `openblas` cc @adamjstewart @tgamblin

Reply:
The problem with openblas > 0.3.5 is that it compiles fine but doesn't pass the blaspp internal tests. We need to prevent people from using this until the devs can resolve this for blaspp. If we instead added a conflicts(), the user would still have to give an explicit version for openblas.

Reply:
Do you think that's an OpenBLAS issue or does BlasPP need to update to account for changes in newer OpenBLAS releases? Since OpenBLAS 0.3.5 does not build on Skylake CPUs (macOS and Linux), using a newer version would really help us.

As a background, we want to combine our app (ECP WarpX) in environments with complex Python stacks, where the scientific Python stack often depends on newer versions of OpenBLAS. So constraining the upper version of OpenBLAS limits our usage of BlasPP for us to build a full-feature stack.

Reply:
We see an issue with OpenBLAS >= 0.3.6, verified through 0.3.13. It shows in our batched BLAS routines, which are doing regular BLAS inside an OpenMP parallel section. Regular BLAS routines are okay, as long as they aren't in an OpenMP parallel section. See https://github.com/xianyi/OpenBLAS/issues/3057

Reply:
Thank you for reporting this upstream, @mgates3.
Looks like we might be able to work-around with [locking](https://github.com/xianyi/OpenBLAS/issues/3057#issuecomment-757559901), let's see where you arrive at with upstream.

Reply:
We have determined that using `^openblas threads=openmp` or `^openblas threads=pthreads` will resolve the problems that we were seeing with openblas versions newer than 0.3.5.  I have tried to add this as a `depends_on('openblas threads=openmp', when='^openblas')`, but it experiences the same problem that we saw when using `depends_on('openblas@:0.3.5', when='^openblas')` (related to the use of `depends_on('blas')`). I also tried a `conflicts('^openblas threads=none)` with no luck.

How is it recommended that we require `openblas threads=openmp` or `openblas threads=pthreads` only when openblas is the chosen blas provider?

Reply:
cc @adamjstewart @alalazo

Reply:
Virtual packages + variants are currently broken, see #1712 and #6300.

As far as I know, the following _should_ work:
```python
depends_on('blas')
depends_on('lapack')
conflicts('^openblas threads=none')
```
That is, it should prevent you from building `^openblas threads=none`, although unless you're using the new concretizer, it won't automatically switch to a different variant that does work, it will just crash.

Reply:
Yes, that looks like the best we can do right now.  I'll create a PR.